python/cudf/cudf/core/udf/__init__.py

```diff
@@ -1,5 +1,6 @@
 # Copyright (c) 2022-2023, NVIDIA CORPORATION.
 from . import (
+    _nrt_cuda,
```

Comment:
Do we need this here?

Reply:
Importing `_nrt_cuda` has side-effects, so it's needed somewhere during import. An argument could be made that the side-effects of its import should be moved into a function (e.g. `init_cuda_nrt()` or whatever) then the function called from this file like `_nrt_cuda.init_cuda_nrt()` so that the effect is more transparent.

This pattern of imports having side-effects that add to Numba's registries or otherwise change / augment it is a bit pervasive in Numba, but it wasn't a good idea and we'd do it differently if we started again - it may be better not to bring that pattern over into cuDF.

Reply:
The important effect here is that `enable_nrt` is set to `True` in the cuda target context at import.